### PR TITLE
Show error if flag is provided multiple times

### LINKED ISSUE #72: Return an error if the same flag is provided twice with different values
In ncw/rclone#506 a user is rightly complaining about the fact that two `--exclude` flags don't work in rclone and it didn't warn him.

Could pflag return an error if you provide the same flag twice with different values?

Thanks

Nick

### PR DESCRIPTION
**Pull request by bogem**
Fix #72 

Also prevent showing message of deprecated or used multiple times flag.

### CODE REVIEW

File: flag.go [hunk 1]
```diff
@@ -828,19 +830,24 @@ func (f *FlagSet) usage() {
 
 func (f *FlagSet) setFlag(flag *Flag, value string, origArg string) error {
 	if err := flag.Value.Set(value); err != nil {
-		return f.failf("invalid argument %q for %s: %v", value, origArg, err)
+		return f.failf("invalid argument %q for %v: %v", value, origArg, err)
+	}
+	if !flag.isSetMultipleTimes && flag.Changed && !strings.Contains(flag.Value.Type(), "Slice") {
+		flag.isSetMultipleTimes = true
+		fmt.Fprintf(os.Stderr, "Flag %q is set multiple times. Using last value\n", flag.Name)
```
Comment (eparis) [inline flag.go]: I think pretty much every flag parser I know of does not consider this an error and none of them warn about it. Do you have an example of a flag parser which complains like this?

Comment (bogem) [inline flag.go]: No, I have no example of such flag parser :(

File: flag.go [hunk 2]
```diff
@@ -828,19 +830,24 @@ func (f *FlagSet) usage() {
...
+
 	// mark as visited for Visit()
 	if f.actual == nil {
 		f.actual = make(map[NormalizedName]*Flag)
 	}
 	f.actual[f.normalizeFlagName(flag.Name)] = flag
 	f.orderedActual = append(f.orderedActual, flag)
 	flag.Changed = true
-	if len(flag.Deprecated) > 0 {
+	if len(flag.Deprecated) > 0 && !flag.isSetMultipleTimes {
```
Comment (eparis) [inline flag.go]: These parts seems interesting though, even if we don't take the generic warning...

**Comment by bogem** [comment]
Yes, it's not a good idea to add this fail message. I didn't found any CLI application that shows an error on flags, which set multiple times. Then we should discuss in #72, how this issue should be handled

File: flag.go (hunk 2 above)
Comment (bogem) [inline flag.go]: Yes, I also thought about what. pflag will warn multiple times about deprecated flag, if user sets deprecated flag multiple times. 
Should I add this part in separate PR?

**Pull request closed without merging**